chaco/tests/test_line_scatterplot.py

```diff
@@ -155,3 +155,26 @@ def test_scatter_1d_selection_alpha(self):
         gc.render_component(self.scatterplot)
         actual = gc.bmp_array[:, :, :]
         self.assertFalse(alltrue(actual == 255))
+
+    def test_scatter_1d_set_index_range(self):
+        new_range = DataRange1D(low=0.42, high=1.42)
+        self.scatterplot.index_range = new_range
+        self.assertEqual(self.scatterplot.index_mapper.range, new_range)
+
+    def test_scatter_1d_set_index_mapper_notifies_index_range(self):
+        new_range = DataRange1D(low=0.42, high=1.42)
+
+        # Given
+        self.notification_has_fired = False
+
+        def _listener():
+            self.notification_has_fired = True
+
+        self.scatterplot.on_trait_change(_listener, "index_range")
```

Comment:
You can handle this more easily with
```
with self.assertTraitChanges(self.scatterplot, "index_range", count=1):
    self.scatterplot.index_mapper = LinearMapper(range=new_range)
```
but you will need to change the test case class to also inherit from the `UnittestTools` mixin class from `traits.testing.unittest_tools`.